src/components/DateRelated.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { StyleSheet, View, Text } from 'react-native';
+import colors from '@styles/colors';
+import { format } from 'date-fns';
+import { ru } from 'date-fns/locale';
+import { WeatherIcon, WeatherType } from '@components/WeatherIcon';
+
+interface IProps {
+    dayOfTheWeek: string;
+    date: string;
+    weatherType: WeatherType;
+    temperatureAfternoon: number;
+    temperatureNight: number;
+}
+
+const getWeek = format(new Date(), 'cccc', { locale: ru });
+
+export const DateRelated: React.FC<IProps> = props => {
+    return (
+        <View style={styles.container}>
+            <View>
+                <Text style={styles.nameWeek}>{props.dayOfTheWeek === getWeek ? 'Сегодня' : props.dayOfTheWeek}</Text>
+                <Text style={styles.numberDate}>{props.date}</Text>
+            </View>
```

Comment:
As we get date in ISO8601 string, you should get all necessary values from it: get Date instance of it, format it to display localed date